src/main/java/webserver/Setup.java

```diff
@@ -0,0 +1,27 @@
+package webserver;
+
+public class Setup {
+    private final int DEFAULT_PORT = 5000;
+    private final String ENV_PORT;
+
+    public Setup(final SystemUtils system) {
```

Comment:
This class looks good!  I like that parsing and applying defaults are handled in one, separate place like this.

A couple nitpicks:

* `final` is great, but you don't often have to use it for parameters like this.  It's used by far the most often for field declarations (where it should be used habitually, by default) and sometimes for local variables.  But typically people expect that if they call a function (or constructor), it's not going to do anything weird to the parameters that are passed in (Java also has a ["pass by value, by reference" behavior](https://stackoverflow.com/questions/373419/whats-the-difference-between-passing-by-reference-vs-passing-by-value); TL;DR you can't re-assign a parameter inside a function and expect it to be re-assigned outside of the function when it returns)
* The field `ENV_PORT` by convention should be more like `envPort` because it's not a constant.